### PR TITLE
Issue: #4042: Improvement for spline option doc for resampling

### PR DESCRIPTION
**Pull request by rutujasurve94**
Added documentation for 'spline' options defining the mapping to new co-ordinates for re-sampling and use of 'spline' for interpolation using piecewise polynomials. Also mentioning the function used for doing the same.

### CODE REVIEW

File: changelog/4136.splinedoc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improvement for spline option doc for resampling.
```
Comment (nabobalis) [inline changelog/4136.splinedoc.rst:1]:
```suggestion
Clarified spline option for `sunpy.map.Map.resampling`.
```
This file should also be renamed `4136.docfix.rst`

Comment (rutujasurve94) [reply changelog/4136.splinedoc.rst:1]: Done

Comment (nabobalis) [reply changelog/4136.splinedoc.rst:1]: Thanks, can you also delete the old file?

Comment (rutujasurve94) [reply changelog/4136.splinedoc.rst]: Yes, deleted the same.

**Comment by nabobalis** [comment]
Thanks @rutujasurve94

**Comment by rutujasurve94** [comment]
Hi, could you also review the doc changes for #4135 (solar cycle example) as well?

**Pull request merged**